src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java

```diff
@@ -213,25 +222,62 @@ private void toggleBooksGoalInfo(boolean isOn, boolean hasReachedGoal) {
     }
 
     /**
-     * Calculates the reading progress for the books goal only
-     * @param booksToReadThisYear the number of books to read by the end of the year (the goal)
-     * @param booksReadThisYear the number of books that have already been read by th end of the year
-     * @return a String that displays whether the goal was met, or whether the user is ahead or behind schedule
+     * @param isOn if true, set the visibility of the page goal-specific text to true. Otherwise, set them to false
+     * @param hasReachedGoal if true, set the visibility of the "books to read on average" text to false.
      */
-    String calculateProgress(int booksToReadThisYear, int booksReadThisYear) {
-        LOGGER.log(Level.INFO, "\nBooks to read this year: " + booksToReadThisYear);
-        LOGGER.log(Level.INFO, "Books read this year: " + booksReadThisYear);
+    private void togglePagesGoalInfo(boolean isOn, boolean hasReachedGoal) {
+        goalProgress.setVisible(isOn);
+        pagesToRead.setVisible(isOn && !hasReachedGoal);
+    }
 
+    /**
+     * Calculates the reading progress for the books/pages goal
+     * @param toReadThisYear the number of books/pages to read by the end of the year (the goal)
+     * @param readThisYear the number of books/pages that have already been read by th end of the year
+     * @return a String that displays whether the goal was met, or whether the user is ahead or behind schedule
+     */
+    String calculateProgress(int toReadThisYear, int readThisYear, ReadingGoal.GoalType goalType) {
         String schedule = "";
-        int booksStillToRead = booksToReadThisYear - booksReadThisYear;
-        LOGGER.log(Level.INFO, "Books still to read: " + booksStillToRead);
+        int stillToRead = toReadThisYear - readThisYear;
 
-        if (booksStillToRead <= 0) {
-            schedule = TARGET_MET;
-        } else {
-            int howManyBehindOrAhead = howFarAheadOrBehindSchedule(booksToReadThisYear, booksReadThisYear);
-            schedule = String.format("You are %d "+ getPluralized("book", howManyBehindOrAhead) + " %s schedule",
-                    howManyBehindOrAhead, behindOrAheadSchedule(booksReadThisYear, shouldHaveRead(booksToReadThisYear)));
+        if (goalType.equals(ReadingGoal.GoalType.BOOKS)) {
```

Comment:
Now that the progress is also being shown for a pages goal, this `if` block looks very similar to the next `if` block. Could you refactor this to remove the duplication (e.g. extract the commonality into a new method(s))?